### ISSUE TITLE
RunInBackground flag does not work

### ISSUE THREAD

**Issue by mehmetkemalbayer** [aws-amplify/aws-sdk-android#668]
Why does runinbackground get network in main thread error on android?
**Which AWS Services are you utilizing?**
AWS Cognito user.initiateUserAuthentication
**Provide code snippets (if applicable)**
    
```
AuthenticationDetails authenticationDetails = new AuthenticationDetails(phoneNumber, new HashMap<String, String>(), new HashMap<String, String>());
                String secretHash = CognitoSecretHash.getSecretHash(phoneNumber, clientId, clientSecret);
                authenticationDetails.setAuthenticationParameter(CognitoServiceConstants.AUTH_PARAM_USERNAME, phoneNumber);
                authenticationDetails.setAuthenticationParameter(CognitoServiceConstants.AUTH_PARAM_SECRET_HASH, secretHash);

        user.initiateUserAuthentication(authenticationDetails, new AuthenticationHandler() {
            @Override
            public void onSuccess(CognitoUserSession userSession, CognitoDevice newDevice) {
                Log.d(TAG, "Success");
            }

            @Override
            public void getAuthenticationDetails(AuthenticationContinuation authenticationContinuation, String userId) {
                Log.d(TAG, "details");
            }

            @Override
            public void getMFACode(MultiFactorAuthenticationContinuation continuation) {
                Log.d(TAG, "mfa");
            }

            @Override
            public void authenticationChallenge(ChallengeContinuation continuation) {
                Intent intent = new Intent(context, LoginSmsActivity.class);
                context.startActivity(intent);
            }

            @Override
            public void onFailure(Exception exception) {
                Log.d(TAG, "Failure");
            }
        }, true).run();
```
**Environment(please complete the following information):**
aws-android-sdk-auth-userpools:2.6.+@aar'

**Device Information (please complete the following information):**
Nexus 6 API 24

**Comment by minbi** [comment]
Hi @mehmetkemalbayer ,

We have identified at least one flow that does not respect the flag. Which flow are you using? or could you provide a stacktrace?

**Comment by minbi** [comment]
Hi @mehmetkemalbayer ,

We are addressing a few issues in the custom auth flow in an upcoming release.

For the namesake issue, we are still working on fixing the NetworkOnMainThread issue. If you need an urgent fix you will need to wrap the call in a background thread. We are also open to pull requests.

**Comment by minbi** [comment]
Hi @mehmetkemalbayer ,

We have released version 2.12.0 of the SDK. Please see if this fixes the "user Id cannot be null" error in the custom auth flow. Please stay tune for fixes to the NetworkOnMainThread issue.

**Comment by minbi** [comment]
Hi @mehmetkemalbayer ,

We have fix pending for this issue, please check it out if you are interested on #702
